frontend/src/components/pages/agents/details/a2a/chat/utils/message-converter.ts

```diff
@@ -18,17 +18,31 @@ import type { ChatMessage, ContentBlock } from '../types';
  * Deserialize contentBlocks from database (ISO string → Date)
  */
 const deserializeContentBlocks = (dbBlocks: import('database/chat-db').ContentBlock[]): ContentBlock[] =>
-  dbBlocks.map((block) => {
+  dbBlocks.map((block: any) => {
+    // Handle legacy 'text' and 'status-update' types (convert to 'task-status-update')
```

Comment:
I think we can remove this - if there is a legacy chat, we will just assume the user has to clear history, it's ephemeral anyway for now.

Let's only support the newly created chats